swap_meet/vendor.py

```diff
@@ -1,2 +1,229 @@
+  - This method considers the first item in the instance's `inventory`, and the first item in the friend's `inventory`
+  - It removes the first item from its `inventory`, and adds the friend's first item
+  - It removes the first item from the friend's `inventory`, and adds the instances first item
+  - It returns `True`
+  - If either itself or the friend have an empty `inventory`, the method returns `False`
+
+### Wave 6
+
+In Wave 6 we will write two methods, `get_best_by_category` and `swap_best_by_category`.
+
+- `Vendor`s have a method named `get_best_by_category`, which will get the item with the best condition in a certain category
+  - It takes one argument: a string that represents a category
+  - This method looks through the instance's `inventory` for the item with the highest `condition` and matching `category`
+    - It returns this item
+    - If there are no items in the `inventory` that match the category, it returns `None`
+    - It returns a single item even if there are duplicates (two or more of the same item with the same condition)
+
+The remaining tests in wave 6 imply:
+
+- `Vendor`s have a method named `swap_best_by_category`, which will swap the best item of certain categories with another `Vendor`
+  - It takes in three arguments
+    - `other`, which represents another `Vendor` instance to trade with
+    - `my_priority`, which represents a category that the `Vendor` wants to receive
+    - `their_priority`, which represents a category that `other` wants to receive
+  - The best item in my inventory that matches `their_priority` category is swapped with the best item in `other`'s inventory that matches `my_priority`
+    - It returns `True`
+    - If the `Vendor` has no item that matches `their_priority` category, swapping does not happen, and it returns `False`
+    - If `other` has no item that matches `my_priority` category, swapping does not happen, and it returns `False`
+
+"""
+
+from asyncio.format_helpers import _format_callback_source
+
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    #with .pop()
+    def remove(self, item):
+        for i in range(len(self.inventory)):
+            if self.inventory[i] == item:
+                removed = self.inventory.pop(i)
+                return removed
+        return False
+
+    #with .remove()
+    # def remove(self,item):
+    #     aList = self.inventory
+    #     for thing in aList:
+    #         if thing == item:
+    #             self.inventory.remove(thing)
+    #             return thing
+    #     return False
```

Comment:
While both approaches are valid, the `remove` approach requires less iteration.

For the second approach, we actually wouldn't need to iterate over the `self.inventory` to find the item. `remove` under the hood, performs the iteration to find the value in the list, and then removes it. 

```python
def remove(self, item):
    if item not in self.inventory:
          return False
          
    self.inventory.remove(item)
```

Reply:
Also, I'm curious why did you choose `pop` over `remove`? Was it to avoid the side effects of editing a list during iteration?